### PR TITLE
Implement external converters

### PR DESCRIPTION
**Pull request by nurikk**
Add support for external converters
https://github.com/Koenkk/zigbee-herdsman-converters/issues/1343

### CODE REVIEW

File: lib/extension/externalConverters.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+const Extension = require('./extension');
+const data = require('../util/data');
+
+
+class ExternalConverters extends Extension {
+    constructor(zigbee, mqtt, state, publishEntityState, eventBus) {
+        super(zigbee, mqtt, state, publishEntityState, eventBus);
+
+        const {
+            // eslint-disable-next-line camelcase
```
Comment (Koenkk) [inline lib/extension/externalConverters.js:12]: Can you remove `// eslint-disable-next-line camelcase` and use `const externalConverters = settings.get(). external_converters;` here?

File: test/externalConverters.test.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        expect(zigbeeHerdsmanConverters.addDeviceDefinition).toHaveBeenNthCalledWith(1, {
+            mock: 1
+        });
+        expect(zigbeeHerdsmanConverters.addDeviceDefinition).toHaveBeenNthCalledWith(2, {
+            mock: 2
+        });
+    });
+
+
+
```
Comment (Koenkk) [inline test/externalConverters.test.js:96]: Please fix the new lines, 1 new line after each test statement

File: test/externalConverters.test.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        settings.set(['external_converters'], ['mock-external-converter-module']);
+        controller = new Controller();
+        await controller.start();
+        await flushPromises();
+        expect(zigbeeHerdsmanConverters.addDeviceDefinition).toHaveBeenCalledTimes(1);
+        expect(zigbeeHerdsmanConverters.addDeviceDefinition).toHaveBeenCalledWith({
+            mock: true
+        });
+    });
+    it('Loads multiple external converters from package', async () => {
```
Comment (Koenkk) [inline test/externalConverters.test.js:108]: Add new line here

File: test/externalConverters.test.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        await flushPromises();
+        expect(zigbeeHerdsmanConverters.addDeviceDefinition).toHaveBeenCalledTimes(1);
+        expect(zigbeeHerdsmanConverters.addDeviceDefinition).toHaveBeenCalledWith({
+            mock: true
+        });
+
+    });
+
+    it('Loads multiple external converters', async () => {
+
```
Comment (Koenkk) [inline test/externalConverters.test.js:79]: Remove these new lines (for all test cases)

**Comment by Koenkk** [comment]
Besides some small cosmetics comments I don't have any "real" comments. If it works and the cosmetic issues are fixed this can be merged. Can you also create a PR to update the documentation?

**Comment by nurikk** [comment]
Tested, everything works as expected :)

**Comment by nurikk** [comment]
> Besides some small cosmetics comments I don't have any "real" comments. If it works and the cosmetic issues are fixed this can be merged. Can you also create a PR to update the documentation?

All cosmetic fixes are done, tested. Now I'm writing documentation for this feature

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**